source/fe/fe_tools_interpolate.cc

```diff
@@ -353,10 +285,6 @@ namespace FETools
           ||
           (subdomain_id == numbers::invalid_subdomain_id))
         {
-          Assert(cell->get_fe().n_components() == fe2.n_components(),
-                 ExcDimensionMismatch(cell->get_fe().n_components(),
-                                      fe2.n_components()));
```

Comment:
It's just moved to [line 249](https://github.com/dealii/dealii/pull/3907/files#diff-efe062ada8a2d637a46661f59bcd4452R249). Since the number of components is the [same](https://www.dealii.org/developer/doxygen/deal.II/classhp_1_1FECollection.html#a48575b94550f07e2a3795afd3ab2390f) for all elements in `hp::FECollection`, we don't have to check on each cell.